components/support/rs_client/src/tests/test_http_client.rs

```diff
@@ -0,0 +1,119 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::{http_client::*, config::ClientConfig, error::ClientError};
+use url::Url;
+use mockito::mock;
+use std::cell::Cell;
+use std::time::{Duration, Instant};
+
+#[test]
+fn test_defaults() {
+    let config = ClientConfig {
+        server_url: None,
+        bucket_name: None,
+        collection_name: String::from("the-collection"),
+    };
+    let client = Client::new(config).unwrap();
+    assert_eq!(Url::parse("https://firefox.settings.services.mozilla.com").unwrap(), client.base_url);
+    assert_eq!(String::from("main"), client.bucket_name);
+}
```

Comment:
It doesn't seem like these tests are run using `cargo test` from repo root. Would this component need to included in the top-level `Cargo.toml` for that to work?

Reply:
ah yes, this should be in the "workspace" (ie, referenced in the top-level Cargo.toml)

Reply:
Ah I bet I actually did have it there in my other branch where I created the top-level Remote Settings stubs and then scrubbed it while I was separating this commit. Thanks!